src/csharp/Grpc.Core.Api/DeserializationContext.cs

```diff
@@ -48,6 +96,23 @@ public virtual byte[] PayloadAsNewBuffer()
             throw new NotImplementedException();
         }
 
+        /// <summary>
+        /// Gets the entire payload as a leased byte array.
+        /// The caller is expected to call Dispose exactly once, returning the buffer to the pool. Accessing
+        /// the data after disposal or calling Dispose multiple times will result in undefined behavior.
+        /// Only the range defined by Offset and Count shoud be accessed by the consumer.
+        /// </summary>
+        /// <returns>a leased buffer containing the entire payload</returns>
+        public virtual LeasedBuffer PayloadAsLeasedBuffer()
```

Comment:
Btw, any reason for not using IMemoryOwner instead of defining a custom type?

Reply:
IMemoryOwner doesn't expose an array, but: it is probably not important as I think we can lose this API completely given the context I now understand about ROS-byte